tests/unit/amo/utils/test_index.js

```diff
@@ -13,4 +15,211 @@ describe(__filename, () => {
       );
     });
   });
+
+  describe('makeInternalExternalURL', () => {
+    const baseURL = 'https://example.org';
+    const defaultConfig = getFakeConfig({ baseURL });
+    const initialQuery = { param1: 'param1' };
+    const pathname = '/path/name';
+
+    const _makeInternalExternalURL = ({
+      _config = defaultConfig,
+      externalQueryParams,
+      internalQueryParams,
+      urlString,
+    }) => {
+      return makeInternalExternalURL({
+        _config,
+        externalQueryParams,
+        internalQueryParams,
+        urlString,
+      });
+    };
+
+    it('adds internal query params to a relative URL', () => {
+      const internalQueryParams = { internalParam1: 'internalParam1' };
+      const urlString = url.format({ pathname, query: initialQuery });
+      const expectedURL = url.format({
+        pathname,
+        query: { ...initialQuery, ...internalQueryParams },
+      });
+
+      expect(
+        _makeInternalExternalURL({
+          internalQueryParams,
+          urlString,
+        }),
+      ).toEqual(expectedURL);
+    });
+
+    it('overrides internal query params for a relative URL', () => {
+      const originalQueryParams = { internalParam1: 'internalParam1' };
+      const internalQueryParams = { internalParam1: 'internalParam2' };
+      const urlString = url.format({ pathname, query: originalQueryParams });
+      const expectedURL = url.format({
+        pathname,
+        query: { ...originalQueryParams, ...internalQueryParams },
+      });
+
+      expect(
+        _makeInternalExternalURL({
+          internalQueryParams,
+          urlString,
+        }),
+      ).toEqual(expectedURL);
+    });
+
+    it('adds internal query params to an absolute URL for the host', () => {
```

Comment:
I think I know what this means but it tripped me up and I had to read it several times. I suggest trying to reword it.